operator/keys/keys.go

```diff
@@ -71,7 +69,8 @@ func GeneratePrivateKey() (OperatorPrivateKey, error) {
 }
 
 type privateKey struct {
-	privKey *rsa.PrivateKey
+	privKey       *rsa.PrivateKey
+	cachedPrivKey any
```

Comment:
What if we create another `privateKey` in `keys_linux.go` which contains `*openssl.PrivateKeyRSA` and use the new library there without using `any` type? This would also simplify the code for signing.